File: app/javascript/channels/chatroom_channel.js
import consumer from "channels/consumer"

const appChat = consumer.subscriptions.create("ChatroomChannel", {
  connected() {
    console.log('connected');
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log(data);
    messages.insertAdjacentHTML('afterbegin', data['message']);
  },

  talk(data) {
    return this.perform('talk', data);
  }
});

document.getElementById('content').addEventListener('keyup', (e) => {
  if (e.key === 'Enter') {
    const sender = document.getElementById('sender');
    appChat.talk({sender: sender.value, content: e.target.value});
    e.target.value = '';
    e.preventDefault();
  }
})

File: app/channels/chatroom_channel.rb
class ChatroomChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from "chatroom_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def talk(data)
    message = Message.new
    message.sender = data['sender']
    message.content = data['content']

    ActionCable.server.broadcast("chatroom_channel", {
      message: render_message(message),
    })
  end

  private

  def render_message(message)
    ApplicationController.renderer.render(partial: 'messages/message', locals: { message: message })
  end
end
